src/module.ts

```diff
@@ -26,40 +26,46 @@ export default defineNuxtModule<ModuleOptions>({
     nuxt.options.security = defu(nuxt.options.security, {
       ...options
     })
+    const securityOptions = nuxt.options.security
 
     // Register nitro plugin to replace default 'X-Powered-By' header with custom one that does not indicate what is the framework underneath the app.
-    if (nuxt.options.security.hidePoweredBy) {
+    if (securityOptions.hidePoweredBy) {
       nuxt.hook('nitro:config', (config) => {
         config.plugins = config.plugins || []
         config.plugins.push(normalize(fileURLToPath(new URL('./runtime/nitro', import.meta.url))))
       })
     }
 
     nuxt.options.runtimeConfig.security = defu(nuxt.options.runtimeConfig.security, {
-      ...nuxt.options.security as RuntimeConfig['security']
+      ...securityOptions as RuntimeConfig['security']
     })
 
     // Register enabled middlewares to automatically set default values for security response headers.
-    if (nuxt.options.security.headers) {
-      for (const header in nuxt.options.security.headers as SecurityHeaders) {
-        if ((nuxt.options.security.headers as SecurityHeader)[header]) {
-          // Have to create this manually, otherwise the build will fail. Also, have to create empty object first or use the previous headers if they are for the same route
-          nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route] = { headers: nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route] ? { ...nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route].headers } : {} }
-
-          nuxt.options.nitro.routeRules!![(nuxt.options.security.headers as SecurityHeader)[header].route].headers!![SECURITY_HEADER_NAMES[header]] = (nuxt.options.security.headers as SecurityHeader)[header].value
+    if (securityOptions.headers) {
+      for (const header in securityOptions.headers as SecurityHeaders) {
+        const headerOptions = (securityOptions.headers as SecurityHeader)[header]
+        if (headerOptions) {
+          const nitroRouteRules = nuxt.options.nitro.routeRules
+          nitroRouteRules!![headerOptions.route] = {
+            ...nitroRouteRules!![headerOptions.route],
```

Comment:
Also, added this little fella, as I found the previous version disregarding any potential previous of entire route rules.